pkg/fanal/analyzer/language/python/poetry/poetry.go

```diff
@@ -105,34 +105,68 @@ func (a poetryAnalyzer) mergePyProject(fsys fs.FS, dir string, app *types.Applic
 
 	// Identify the direct/transitive dependencies
 	for i, pkg := range app.Packages {
-		if _, ok := p[pkg.Name]; ok {
+		if _, ok := project.Tool.Poetry.Dependencies[pkg.Name]; ok {
 			app.Packages[i].Relationship = types.RelationshipDirect
 		} else {
 			app.Packages[i].Indirect = true
 			app.Packages[i].Relationship = types.RelationshipIndirect
 		}
 	}
 
+	filterProdPackages(project, app)
 	return nil
 }
 
-func (a poetryAnalyzer) parsePyProject(fsys fs.FS, path string) (map[string]any, error) {
+func filterProdPackages(project pyproject.PyProject, app *types.Application) {
```

Comment:
We should flag [the Dev field](https://github.com/aquasecurity/trivy/blob/d622ca2b1fe40a0eb588478ba9e15d3bd8471a78/pkg/fanal/types/package.go#L187) instead of filtering for `--include-dev-deps`, but if you want to keep the original behavior in this PR, we can add the change in a separate PR.

Reply:
@knqyf263 as i wrote in https://github.com/aquasecurity/trivy/pull/8080#discussion_r1891328554 - are you sure that we want to add dev deps under `Dev` field?
Previously we only did this upon request from users

Reply:
Hmm. It sounds reasonable, but we should add a new column showing if --include-dev-deps is supported.

Reply:
But if the implementation cost is not so different, it's better to mark dev dependencies.

Reply:
I'll open another PR for this one